gcovr/formats/gcov/parser.py

```diff
@@ -453,7 +453,7 @@ def _reconstruct_source_code(tokens: Iterable[_Line]) -> List[str]:
 class _ParserState(NamedTuple):
     deferred_functions: List[_FunctionLine] = []
     lineno: int = 0
-    blockno: int = None
+    blockno: int = 0
```

Comment:
I'm not totally sure if we want to change this default. Are there scenarios where it might be helpful to show in our internal data format (and the JSON report) that we didn't get any block-level coverage data?

I absolutely agree that replacing `None` with `0` is important in the lcov-writer, but that could also be done without changes to our data model.

Reply:
You're right here. Maybe it's better to keep the parser as it is and add a property `ensure_blockno` which returns `0` if it is `None`.
In that case we can also print a info message.